packages/swapper/src/swappers/RelaySwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,242 @@
+      }),
+    )
+  }
+
+  if (!sendAddress) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'sendAddress is required',
+      }),
+    )
+  }
+
+  const maybeQuote = await getQuote(
+    {
+      originChainId: sellRelayChainId,
+      originCurrency: getRelayEvmAssetAddress(sellAsset),
+      destinationChainId: buyRelayChainId,
+      destinationCurrency: getRelayEvmAssetAddress(buyAsset),
+      tradeType: 'EXACT_INPUT',
+      amount: sellAmountIncludingProtocolFeesCryptoBaseUnit,
+      recipient: receiveAddress,
+      user: sendAddress,
+      slippageTolerance: slippageToleranceBps,
+    },
+    deps.config,
+  )
+
+  if (maybeQuote.isErr()) return Err(maybeQuote.unwrapErr())
+
+  const quote = maybeQuote.unwrap()
+
+  const swapSteps = quote.data.steps.filter(step => step.id !== 'approve')
+  const hasApprovalStep = quote.data.steps.find(step => step.id === 'approve')
+
+  if (swapSteps.length > 2) {
+    deps.mixPanel?.track(MixPanelEvent.RelayMultiSteps, {
+      swapper: SwapperName.Relay,
+      method: 'get',
+    })
+
+    return Err(
+      makeSwapErrorRight({
+        message: `Relay quote with ${swapSteps.length} swap steps not supported (maximum 2)`,
+        code: TradeQuoteError.UnsupportedTradePair,
+      }),
+    )
+  }
+
+  const slippageTolerancePercentageDecimal = (() => {
+    if (_slippageTolerancePercentageDecimal) return _slippageTolerancePercentageDecimal
+    const destinationSlippageTolerancePercentageDecimal = bnOrZero(
+      quote.data.details?.slippageTolerance?.destination?.percent,
+    )
+
+    if (destinationSlippageTolerancePercentageDecimal.gt(0)) {
+      return convertBasisPointsToPercentage(destinationSlippageTolerancePercentageDecimal).toFixed()
+    }
+
+    const originSlippageTolerancePercentageDecimal = bnOrZero(
+      quote.data.details?.slippageTolerance?.origin?.percent,
+    )
+
+    if (originSlippageTolerancePercentageDecimal.gt(0)) {
+      return convertBasisPointsToPercentage(originSlippageTolerancePercentageDecimal).toFixed()
+    }
+  })()
+
+  const relayToken = quote.data.fees?.relayerService?.currency
+
+  if (!relayToken) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Relay protocol token not found',
+      }),
+    )
+  }
+
+  if (!isRelayToken(relayToken)) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Relay protocol token is not properly typed',
+      }),
+    )
+  }
+
+  const protocolAssetId = relayTokenToAssetId(relayToken)
+
+  const protocolAsset = relayTokenToAsset(relayToken, deps.assetsById)
+
+  const steps = swapSteps.map(
+    (quoteStep): TradeQuoteStep => ({
+      allowanceContract: hasApprovalStep ? swapSteps[0]?.items?.[0]?.data?.to : undefined,
+      rate: quote.data.details?.rate ?? '0',
+      buyAmountBeforeFeesCryptoBaseUnit: quote.data.details?.currencyOut?.amount ?? '0',
+      buyAmountAfterFeesCryptoBaseUnit: quote.data.details?.currencyOut?.minimumAmount ?? '0',
```

Comment:
Those are not documented but this seems like it could potentially be incorrect, as `minimumAmount` seems like a `MinOut` (guaranteed amount by the solver) which is indeed the amount out we should quite, but `amount` sounds like an optimistic amount (quoted amount which solvers will try to get), i.e not related to fees. Just a bold assumption without docs of course! 

@NeOMakinG can we double-check with the Relay guys what each of these mean? 🙏🏽

Reply:
Happy to ask with the relay team but different between amount and minimumAmount seems to be the right one because it seems to match the :

- Slippage
- Network fees
- Relayer fees

Comparing with the price impact, it looks like being right

Reply:
now that we have the reply from the team, let's make sure to use the min out, same like other swappers

Reply:
It was already the case, but wasn't in the minus fee maths, changed in [29fc44d](https://github.com/shapeshift/web/pull/9188/commits/29fc44d8ce158086f2cbec2c3c7ee0dd44302dac)